Common/BaselineRecoSelection.h

```diff
@@ -163,24 +164,59 @@ RVecI GenRecoJetMatching(int event,const RVecI& Jet_idx, const RVecI& GenJet_idx
 }
 
 
+std::optional<HbbCand> GetHbbCandidate(const RVecF& HHbTagScores, const RVecB& JetSel, const RVecLV& Jet_p4, const RVecI& Jet_idx)
+{
+  RVecI JetIdxOrdered = ReorderObjects(HHbTagScores, Jet_idx);
+  std::optional<HbbCand> HbbCandidate = std::nullopt;  // Inizializza come opzionale senza valore
+
+
+  int leg_idx = 0;
+  RVecI already_taken_idx;
+  for(int i = 0; i < Jet_idx.size(); i++){
+    int jet_idx = JetIdxOrdered[i];
+    if(!JetSel[jet_idx]) continue;
+    if(std::find(already_taken_idx.begin(), already_taken_idx.end(), jet_idx) != already_taken_idx.end()) continue;
+    HbbCandidate->leg_index[leg_idx] = jet_idx;
+    HbbCandidate->leg_p4[leg_idx] = Jet_p4.at(jet_idx);
+    already_taken_idx.push_back(jet_idx);
+    leg_idx++;
+    if(leg_idx == HbbCandidate->n_legs) break;
+  }
```

Comment:
- how can it be already taken? 
- what if there are no enough legs?
- initially, it is `nullopt`; how can you apply `->` to it?